rbac/api/utils.py

```diff
@@ -1,15 +1,52 @@
 """Helper utilities for api module."""
 
+import logging
+
 from django.db import transaction
+from django.shortcuts import get_object_or_404
+from management.role.model import BindingMapping
+from management.tenant_mapping.model import TenantMapping
 from management.utils import account_id_for_tenant
+from management.workspace.model import Workspace
 
 from api.models import Tenant
 
 
+RESOURCE_MODEL_MAPPING = {
+    "workspace": Workspace,
+    "mapping": TenantMapping,
+    "binding": BindingMapping,
+}
+logger = logging.getLogger(__name__)
+
+
 def populate_tenant_account_id():
     """Populate tenant's account id's."""
     tenants = Tenant.objects.filter(account_id__isnull=True).exclude(tenant_name="public")
     for tenant in tenants:
         with transaction.atomic():
             tenant.account_id = account_id_for_tenant(tenant)
             tenant.save()
+
+
+def get_resources(resource, org_id):
+    """Get queryset by org_id."""
+    queryset = RESOURCE_MODEL_MAPPING[resource].objects.all()
+    if resource != "binding" and org_id:
+        tenant = get_object_or_404(Tenant, org_id=org_id)
+        queryset = queryset.filter(tenant=tenant)
```

Comment:
Maybe abort or something if its the binding? Just so we have to be intentional if we want to delete everything, and someone doesn't think they're only deleting bindings for one org and end up accidentally deleting everything.

Reply:
Updated